dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -865,6 +882,149 @@ public DagDataSchedule exportProcessDagData(ProcessDefinition processDefinition)
         return result;
     }
 
+    @Override
+    public Map<String, Object> importSqlProcessDefinition(User loginUser, long projectCode, MultipartFile file) {
+        Map<String, Object> result = new HashMap<>();
+        String processDefinitionName = file.getOriginalFilename() == null ? file.getName() : file.getOriginalFilename();
+        int index = processDefinitionName.lastIndexOf(".");
+        if (index > 0) {
+            processDefinitionName = processDefinitionName.substring(0, index);
+        }
+        // build process definition
+        Date now = new Date();
+        ProcessDefinition processDefinition = new ProcessDefinition();
+        processDefinition.setName(processDefinitionName);
+        processDefinition.setCreateTime(now);
+        processDefinition.setUpdateTime(now);
+        processDefinition.setFlag(Flag.YES);
+        processDefinition.setTenantId(-1);
+        processDefinition.setGlobalParamList(Collections.emptyList());
+
+        DagDataSchedule dagDataSchedule = new DagDataSchedule();
+        dagDataSchedule.setProcessDefinition(processDefinition);
+        List<TaskDefinition> taskDefinitionList = new ArrayList<>();
+        dagDataSchedule.setTaskDefinitionList(taskDefinitionList);
+        List<ProcessTaskRelation> processTaskRelationList = new ArrayList<>();
+        dagDataSchedule.setProcessTaskRelationList(processTaskRelationList);
+
+        // In most cases, there will be only one data source
+        Map<String, DataSource> dataSourceCache = new HashMap<>(1);
+        Map<String, Long> taskNameToCode = new HashMap<>(16);
+        Map<String, List<String>> taskNameToUpstream = new HashMap<>(16);
+        try (ZipInputStream zIn = new ZipInputStream(file.getInputStream());
+             BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(zIn))) {
+            ZipEntry entry;
+            while ((entry = zIn.getNextEntry()) != null) {
+                if (!entry.isDirectory()) {
+                    StringBuilder sql = new StringBuilder();
+                    String taskName = null;
+                    String datasourceName = null;
+                    List<String> upstreams = Collections.emptyList();
+                    String line;
+                    while ((line = bufferedReader.readLine()) != null) {
+                        int commentIndex = line.indexOf("--");
```

Comment:
I think SQL regular may also contain these symbols, and you should exclude this possibility here. You can refer to
![image](https://user-images.githubusercontent.com/42576980/145762497-3d87f151-b2c8-408f-8fd3-f96155130801.png)